spacesuit/src/bit_range.rs

```diff
@@ -0,0 +1,23 @@
+use bulletproofs::r1cs::R1CSError;
+/// Represents a usize with value in the range [0,64]
+pub struct BitRange(usize);
+
+impl BitRange {
+    /// Returns BitRange if n is <= 64>
+    /// Otherwise Errors
+    pub fn new(n: usize) -> Result<Self, R1CSError> {
```

Comment:
R1CSError is from another framework, not from spacesuit. For such simple cases when there is only one kind of error (see also [`Scalar::from_canonical_bytes`](https://github.com/dalek-cryptography/curve25519-dalek/blob/master/src/scalar.rs#L227-L237)) it is customary to return `Option` instead of `Result`.

One can always convert a None to an error in a higher-level code via `.ok_or(SomeError)?`